s3transfer/upload.py

```diff
@@ -287,8 +289,12 @@ def _get_upload_part_fileobj_with_full_size(self, fileobj, **kwargs):
         # source. So we must treat it as its own standalone file.
         return six.BytesIO(data), len(data)
 
-    def _get_put_object_fileobj(self, fileobj):
-        return fileobj
+    def _get_put_object_fileobj_with_full_size(self, transfer_future):
+        fileobj = transfer_future.meta.call_args.fileobj
+        # The current position needs to be taken into account when retrieving
+        # the full size of the file.
+        size = fileobj.tell() + transfer_future.meta.size
```

Comment:
Yeah that makes sense, thanks for clarifying.

I wish there was a better way to abstract out this detail, but I don't know if it makes sense to move into `ReadFileChunk`.